Выполнение ТЗ-12

klv2503 opened the pull request:
Реализована работа с БД. Ее проект в Readme. На всякий случай оставил реализацию InMemory, дополнив ее работой с новыми сущностями. Тесты InMemory убрал, тесты DB реализованы.

klv2503 committed:
Выполнение ТЗ-12

klv2503 committed:
Update and rename inMemoryRatingStorage.java to InMemoryRatingStorage.java

По недосмотру класс оказался именован с маленькой буквы. Осталось незамеченным, т.к. класс в процессе выполнения ТЗ не использовался

klv2503 committed:
Update FilmDbFilmoRateApplicationTests.java

Удаляем неиспользуемый import

klv2503 committed:
Update application.properties

Поменял never на always

klv2503 committed:
Update UserRequest.java

Добавил @RequiredArgsConstructor

db-exp commented on src/test/resources/schema.sql:
```diff
@@ -0,0 +1,47 @@
+
+CREATE TABLE IF NOT EXISTS genre (
+              id BIGINT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+              name varchar NOT NULL,
```
нужно везде проставить размерность varchar и это не всегда 255)

klv2503 replied:
Ограничения на длину наименования жанров, рейтингов и фильмов указал исходя из самых длинных на сейчас. Description во всех случаях ограничил 200. Users трогать не стал - люди слишком изобретательны, самое длинное имя оказалось длиной под полторы тысячи букв. По итогам внес в Request'ы аннотации. Модификаторы указал, уже и сам какое-то время назад понял, что их расстановка у меня слабое место, пока не выправился. Модификаторы потребовали внести небольшие изменения в несколько классов. Тесты постмана проходятся.

db-exp replied:
по правилам ревью - везде должна быть размерность для varchar

klv2503 replied:
Если я правильно понял, длина email может быть (по разным данным) 254 или 256 символов. Решил ограничиться чем-то разумным. Также ограничился более-менее разумной длиной логина. Что касается длины имени. Оказалось, что полное имя Пикассо на русском записывается строкой в 155 символов (на испанском короче, но не сильно). Ну я и поставил с запасом, кто их, этих испанцев знает... Не ставить же полторы тысячи.

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,47 @@
+
+CREATE TABLE IF NOT EXISTS genre (
+              id BIGINT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+              name varchar NOT NULL,
```
везде нужна размерность для varchar

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/Rating.java:
```diff
@@ -0,0 +1,19 @@
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class Rating {
+    Long id;
```
private final

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/Rating.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+public class Rating {
+    Long id;
+    String name;
```
private

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/FilmGenre.java:
```diff
@@ -0,0 +1,14 @@
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class FilmGenre {
+    Long id;
```
private модификаторы везде потеряны

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dto/GenreRequest.java:
```diff
@@ -0,0 +1,15 @@
+@RequiredArgsConstructor
+public class GenreRequest {
+
+    @NotBlank(message = "Признак жанра не может быть пустым.")
```
к NotBlank всегда можно Size добавлять

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,76 +1,88 @@
-    public Collection<Film> findFilms(@RequestParam(required = false, defaultValue = "10") @Positive long count) {
-        log.info("\nGetting {} most popular films", count);
-        return filmService.getPopularFilms(count);
+    Gson gson = new Gson();
```
private final

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dto/FilmRequest.java:
```diff
@@ -0,0 +1,51 @@
+    Long id;
+
+    @NotBlank(message = "Название фильма не может быть пустым")
+    @NotNull(message = "Название фильма не может быть пустым")
```
NotNull и NotBlank можно не ставить вместе, лучше только NotBlank

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/mapper/FilmMapper.java:
```diff
@@ -0,0 +1,51 @@
+
+import java.util.List;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
```
ну или вот так, где private не хватает

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/mapper/FilmMapper.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.List;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+
```
лишний отступ

klv2503 committed:
Выполнение ТЗ-12. Учет замечаний

klv2503 committed:
Merge remote-tracking branch 'origin/add-database' into add-database

klv2503 committed:
Update FilmController.java

При выполнении ТЗ-12 была проблема с десериализацией. Gson использовался в одном из вариантов. Теперь он лишний, его удалил, забыл удалить его импорт.

klv2503 committed:
Update RatingDbFilmorateApplicationTests.java

Укоротил значения строк из-за которых не проходили тесты RatingDbStorage

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dto/GenreRequest.java:
```diff
@@ -0,0 +1,18 @@
+
+    @NotBlank(message = "Признак жанра не может быть пустым.")
+    @Size(message = "Слишком длинное наименование жанра (более 25 символов)", max = 25)
+    String name;
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dal/mappers/RatingRowMapper.java:
```diff
@@ -0,0 +1,22 @@
+
+@Component
+public class RatingRowMapper implements RowMapper<Rating> {
+    Long id;
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dto/FilmDto.java:
```diff
@@ -0,0 +1,40 @@
+public class FilmDto {
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    Long id;
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/dto/UserRequest.java:
```diff
@@ -0,0 +1,44 @@
+public class UserRequest {
+
+    @Positive(message = "Users id must be positive")
+    Long id;
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,42 +1,41 @@
 public class Film {
-
+    //Поскольку контроллер теперь принимает FilmRequest, аннотации перенесены в этот класс
     Long id;
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -17,19 +12,15 @@
 @AllArgsConstructor
 public class User {
+    //Аннотации перенесены в класс UserRequest
     Long id; // уникальный идентификатор пользователя,
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmGenre/InMemoryFilmGenreStorage.java:
```diff
@@ -0,0 +1,46 @@
+@Component("inMemoryFilmGenreStorage")
+public class InMemoryFilmGenreStorage implements FilmGenreStorage {
+
+    Map<Long, FilmGenre> filmGenres = new HashMap<>();
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/storage/rating/InMemoryRatingStorage.java:
```diff
@@ -0,0 +1,61 @@
+@Component("inMemoryRatingStorage")
+public class InMemoryRatingStorage implements RatingStorage {
+
+    Map<Long, Rating> ratingMap = new HashMap<>();
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/test/java/ru/yandex/practicum/filmorate/FilmDbFilmoRateApplicationTests.java:
```diff
@@ -0,0 +1,142 @@
+
+    private final FilmDbStorage filmStorage;
+
+    FilmRequest standartFilm = new FilmRequest("Halloween", "Classic horror",
```
модификаторы доступа все еще отсутствуют)

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,46 @@
+
+CREATE TABLE IF NOT EXISTS users (
+              id BIGINT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+              email varchar NOT NULL,
```
varchar без размерности - нужно проставить везде и в схеме тестов тоже

db-exp commented on src/test/java/ru/yandex/practicum/filmorate/UserDbFilmoRateApplicationTests.java:
```diff
@@ -0,0 +1,180 @@
+class UserDbFilmoRateApplicationTests {
+    private final UserDbStorage userStorage;
+
+    User standartUser1 = new User("first@error.com", "firstLogin", "NameTest1",
```
нет модификатора доступа

db-exp commented on src/test/java/ru/yandex/practicum/filmorate/UserDbFilmoRateApplicationTests.java:
```diff
@@ -0,0 +1,180 @@
+
+    User standartUser1 = new User("first@error.com", "firstLogin", "NameTest1",
+            LocalDate.of(2000, 10, 10));
+    User standartUser2 = new User("second@error.com", "secondLogin", "NameTest2",
```
нет модификатора доступа

klv2503 committed:
Выполнение ТЗ-12. Учет замечания о пользователе.

klv2503 committed:
Merge remote-tracking branch 'origin/add-database' into add-database

klv2503 committed:
Добавлены модификаторы

klv2503 committed:
Добавлены модификаторы в классах тестирования
